components/arm/client.go

```diff
@@ -126,3 +126,11 @@ func (c *client) GoToInputs(ctx context.Context, goal []referenceframe.Input) er
 func (c *client) DoCommand(ctx context.Context, cmd map[string]interface{}) (map[string]interface{}, error) {
 	return generic.DoFromConnection(ctx, c.conn, c.name, cmd)
 }
+
+func (c *client) IsMoving(ctx context.Context) (bool, error) {
```

Comment:
As noted, this is unfortunately needed to make things work for this PR. I do agree it would feel a lot more correct if split out, but it'd then have to be completed and merged first. Happy to discuss. Maybe another team can grab it go do it properly (and quickly) while I/we work through other stuff in this PR?